### DISCUSSION TITLE
MariaDB randomly shuts down - will not start up again

### DISCUSSION THREAD

**Discussion by bernsel** [photoprism/photoprism#1191] [Q&A] [answered]
I am running Photoprism on a Qnap with the help of docker-compose. It all works great, all images indexed, I can access web interface and browse around. But after a few hours of uptime, the database seems to shut down on its own. After that, it will not start up again, when I run docker-compose. See output from log file further down for shutdown event - the shutdown happens at time 07:49:54.

When I try to restart Photoprism (at time 10:16:24 in the log below), it cannot access the MariaDB server. The reason seems to be that when I run "docker-compose up -d photoprism" from the correct folder, it only tries to start the webserver, and not the database. See output below - where normally you would see both "photoprism_mariadb_1" and "photoprism_photoprism_1" starting, but now I only see one of them.

[/share/Containers/photoprism] # docker-compose up -d photoprism
Starting photoprism_photoprism_1 ... done
[/share/Containers/photoprism] # 

I am using version 3.5 of the "docker-compose.yml" provided for Photoprism, and have only change folders and password. Any ideas of what might be wrong?

[/share/Containers/photoprism] # docker-compose logs --tail=25 -f
Attaching to photoprism_photoprism_1, photoprism_mariadb_1
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: 128 rollback segments are active.
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: 10.5.9 started; log sequence number 622946976; transaction id 2226997
mariadb_1     | 2021-04-10 21:06:21 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mariadb_1     | 2021-04-10 21:06:21 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1     | 2021-04-10 21:06:22 0 [Note] Server socket created on IP: '::'.
mariadb_1     | 2021-04-10 21:06:22 0 [Warning] 'proxies_priv' entry '@% root@3104a20c7382' ignored in --skip-name-resolve mode.
mariadb_1     | 2021-04-10 21:06:22 0 [Note] Reading of all Master_info entries succeeded
mariadb_1     | 2021-04-10 21:06:22 0 [Note] Added new Master_info '' to hash table
mariadb_1     | 2021-04-10 21:06:22 0 [Note] mysqld: ready for connections.
mariadb_1     | Version: '10.5.9-MariaDB-1:10.5.9+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
mariadb_1     | 2021-04-10 21:06:22 0 [Note] InnoDB: Buffer pool(s) load completed at 210410 21:06:22
mariadb_1     | 2021-04-11  7:49:54 0 [Note] mysqld (initiated by: unknown): Normal shutdown
mariadb_1     | 2021-04-11  7:49:54 0 [Note] Event Scheduler: Purging the queue. 0 events
mariadb_1     | 2021-04-11  7:49:54 0 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1     | 2021-04-11  7:49:54 0 [Note] InnoDB: Starting shutdown...
mariadb_1     | 2021-04-11  7:49:54 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
mariadb_1     | 2021-04-11  7:49:54 0 [Note] InnoDB: Restricted to 2016 pages due to innodb_buf_pool_dump_pct=25
mariadb_1     | 2021-04-11  7:49:54 0 [Note] InnoDB: Buffer pool(s) dump completed at 210411  7:49:54
mariadb_1     | 2021-04-11  7:49:55 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
mariadb_1     | 2021-04-11  7:49:55 0 [Note] InnoDB: Shutdown completed; log sequence number 631269746; transaction id 2296724
mariadb_1     | 2021-04-11  7:49:55 0 [Note] mysqld: Shutdown complete
mariadb_1     |
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC06478 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC06521 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC06722 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC06900 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC06989 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo DSC07000 as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo YasirCarolinavalima as deleted"
photoprism_1  | time="2021-04-10T21:23:10Z" level=info msg="purge: flagged photo carolinavalima as deleted"
photoprism_1  | time="2021-04-10T21:23:11Z" level=info msg="purge: searching index for unassigned primary files"
photoprism_1  | time="2021-04-10T21:23:11Z" level=info msg="purge: searching index for hidden media files"
photoprism_1  | time="2021-04-10T21:23:15Z" level=info msg="Removed 28 files and 30 photos"
photoprism_1  | time="2021-04-10T21:23:18Z" level=info msg="Indexing completed in 673 s"
photoprism_1  | time="2021-04-11T07:49:54Z" level=info msg="shutting down..."
photoprism_1  | time="2021-04-11T07:49:54Z" level=info msg="shutting down workers"
photoprism_1  | time="2021-04-11T07:49:54Z" level=info msg="closed database connection"
photoprism_1  | time="2021-04-11T07:49:54Z" level=info msg="http: shutting down web server"
photoprism_1  | running as uid 0
photoprism_1  | photoprism start
photoprism_1  | time="2021-04-11T10:02:47Z" level=fatal msg="dial tcp: lookup mariadb on 127.0.0.11:53: no such host"
photoprism_1  | running as uid 0
photoprism_1  | photoprism start
photoprism_1  | time="2021-04-11T10:06:58Z" level=fatal msg="dial tcp: lookup mariadb on 127.0.0.11:53: no such host"
photoprism_1  | running as uid 0
photoprism_1  | photoprism start
photoprism_1  | time="2021-04-11T10:16:24Z" level=fatal msg="dial tcp: lookup mariadb on 127.0.0.11:53: no such host"
photoprism_mariadb_1 exited with code 0
photoprism_photoprism_1 exited with code 0
[/share/Containers/photoprism] #

**Comment by alexisLefebvre** [comment, marked as answer]
> r. The reason seems to be that when I run "docker-compose up -d photoprism" from the correct folder, it only tries to start the webserver, and not the database.

The arguments after `-d` are the names of the services that will be restarted, so here you are restarting only photoprism. Run `docker-compose up -d` to start the 2 services, photoprism **and** the database.

**Comment by bernsel** [comment]
Thank you so much!, that solved it - silly mistake by me.

I am still wondering why both Photoprism services shut down though, at 7:49 AM? If anyone has any theory about that, it would be welcome as well. I am wondering if it might be related to this message in the log "Restricted to 2016 pages due to innodb_buf_pool_dump_pct=25" ? Maybe the database hits some kind of limit, and shuts down because of that?

**Comment by alexisLefebvre** [comment]
The messages `mysqld: Shutdown complete` `http: shutting down web server` indicates that something stopped the containers, it doesn't look like a crash.

**Comment by FreshMosh** [comment]
Agree with @alexislefebvre. 
According to this line
`mariadb_1 | 2021-04-11 7:49:54 0 [Note] mysqld (initiated by: unknown): Normal shutdown`
it is a regular shutdown. 
It might be triggered by restarting the Docker-daemon, the stack or even the enitre host server. 

Also, keep in mind to check in which timezone your server resides. 
Because 7:49am on your server might actually match your working hours, depending on where you are and where the server is located.

**Comment by lastzero** [comment]
You can tell Docker (Compose) to automatically restart if not intentionally stopped. This is documented but commented out in our example config files. You can also change this to always restart even if stopped, which is more aggressive.

**Comment by ankolbi** [comment]
I had the same problem (on a Qnap NAS). The problem was that an old image of latest was used. I had to delete and pull new image.
